pkg/detector/detector.go

```diff
@@ -361,63 +368,57 @@ func (d *ResourceDetector) LookForMatchedPolicy(object *unstructured.Unstructure
 	}
 
 	klog.V(2).Infof("Attempts to match policy for resource(%s)", objectKey)
-	policyObjects, err := d.propagationPolicyLister.ByNamespace(objectKey.Namespace).List(labels.Everything())
+	policyList := &policyv1alpha1.PropagationPolicyList{}
+	err := d.Client.List(context.TODO(), policyList, &client.ListOptions{
+		Namespace:             objectKey.Namespace,
+		UnsafeDisableDeepCopy: ptr.To(true),
```

Comment:
It would be nice to add some comments here to prevent changes to the `policyList` during subsequent iterations.

Reply:
Controller-runtime has a detailed description of `UnsafeDisableDeepCopy`, which is very clear. Is there no need to add it?

https://github.com/karmada-io/karmada/blob/65e7f74c991631a5888f5176342af118f4d5712e/vendor/sigs.k8s.io/controller-runtime/pkg/client/options.go#L495-L499

Reply:
How about not setting this value in the current pr?